### DISCUSSION TITLE
v9.0.0-alpha.0

### DISCUSSION THREAD

**Discussion by thymikee** [callstack/react-native-testing-library#862] [General]
## What's Changed
* feat(breaking): add text match options a.k.a string precision API by @AugustinLF in https://github.com/callstack/react-native-testing-library/pull/554
   Please refer to #554 to see what's breaking

**Full Changelog**: https://github.com/callstack/react-native-testing-library/compare/v8.0.0...v9.0.0-alpha.0

<hr /><em>This discussion was created from the release <a href='https://github.com/callstack/react-native-testing-library/releases/tag/v9.0.0-alpha.0'>v9.0.0-alpha.0</a>.</em>

**Comment by AugustinLF** [comment]
I've tested upgrading out internal codebase to this version, no big surprise/issue. Hard to estimate how many tests we have running this code, but I've counted ~80 files.

For people who rely on nested matching (we had a couple of tests like that), I've seen that `within` is usually a good solution: to match `<Text>Search for <Text>Nike</Text></Text>`, instead of the current `getByText('Search for Nike')` you can write `within(getByText('Search for', {exact: false})).getByText('Nike')`. Obviously not as convenient, but since we're breaking by design, at least there's a fallback.

**Reply by thymikee** [reply]
Thank you!

**Reply by AugustinLF** [reply]
What's your plan regarding a full fledged release? I won't merge a non stable release to our main repo, so if there's anything I can do to help promote this release, please let me know!

**Reply by thymikee** [reply]
It involves breaking changes, so I'd like to create a blog post about the change, its origin and how to migrate. Once we have that, we can bump to stable

**Reply by AugustinLF** [reply]
Yes, I understand, let me know if you need any help on that then!

**Reply by thymikee** [reply]
The post should be pretty short and since you have the context already, maybe you'd be interested in contributing it to the website? It could be yet another migration guide (we have 2 at the moment). I can help with reviewing pushing it over the finishing line :)
